src/lib.rs

```diff
@@ -18,10 +18,23 @@
 use byteorder::ReadBytesExt;
 use std::error::Error;
 use std::fmt::{self, Display};
-use std::io::{Cursor, Read};
+use std::io::{Cursor, Read, Seek, SeekFrom};
 use std::time;
 use zeroize::Zeroize;
 
+#[cfg(feature = "rustcrypto")]
+use aes::cipher::{block_padding::Pkcs7, BlockDecryptMut, BlockEncryptMut, KeyIvInit};
+#[cfg(feature = "rustcrypto")]
+use hmac::Mac;
+#[cfg(feature = "rustcrypto")]
+use sha2::Sha256;
+
+#[cfg(not(any(feature = "rustcrypto", feature = "default",)))]
+compile_error!(
+    "no crypto cargo feature enabled! \
+     please enable one of: default, rustcrypto"
```

Comment:
This is technically fine, but I think `openssl` should be referenced rather than `default`. This is also what the dalek library does

Reply:
Ah, for some reason I thought it wouldn't work since openssl isn't a "feature", but a dependency, but dependencies are implicitly a feature. Fixed.